### PR TITLE
Support samesite option in AuthtktAuthenticationPolicy and CookieCSRFStoragePolicy

### PR DESCRIPTION
**Pull request by mcdonc**
See #2733

### CODE REVIEW

**Comment by stevepiercy** [comment]
It would be good to add a mention of what are same site cookies, why a developer would want to use them, and what the setting does in the narrative documentation. Perhaps in [Security](https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/security.html)?

Are we going to support Lax only, or Strict, too?
https://github.com/Pylons/pyramid/issues/2733#issuecomment-238455333

Finally perhaps linking to the change in WebOb would be useful?
https://github.com/Pylons/pyramid/issues/2733#issuecomment-328362821

File: pyramid/authentication.py [hunk 1]
```diff
@@ -565,8 +565,16 @@ class AuthTktAuthenticationPolicy(CallbackAuthenticationPolicy):
         steps.  The output from debugging is useful for reporting to maillist
         or IRC channels when asking for support.
 
+    ``samesite``
+
+        Default: ``'Lax'``.  The 'samesite' option of the session cookie. Set
+        the value to ``None`` to turn off the samesite option.
+
+        This option is available as of :app:`Pyramid` 1.10.
```
Comment (mmerickel) [inline pyramid/authentication.py]: Move the version info down to a `.. versionchanged:: 1.10`` directive at the bottom of the docstring and document it as "Added the ``samesite`` option and made the default ``'Lax'``." similar to below in the csrf storage policy.

File: CHANGES.rst [hunk 2]
```diff
@@ -37,6 +37,10 @@ Features
   ``pyramid.session.UnencryptedCookieSessionFactoryConfig``.
   See https://github.com/Pylons/pyramid/pull/3300
 
+- Modify ``pyramid.authentication.AuthTktAuthenticationPolicy`` and
+  ``pyramid.csrf.CookieCSRFStoragePolicy`` to support the SameSite option on
+  cookies.  See https://github.com/Pylons/pyramid/pull/3319
+
```
Comment (mmerickel) [inline CHANGES.rst]: Document that the default value was changed to ``Lax``.

**Comment by mmerickel** [comment]
@stevepiercy 

> It would be good to add a mention of what are same site cookies, why a developer would want to use them, and what the setting does in the narrative documentation. Perhaps in Security?

This is fine but I don't think it's required for this PR. That has more to do with speculative best practices and hardening like https://github.com/Pylons/pyramid/issues/1845 is requesting.

> Are we going to support Lax only, or Strict, too?

lax/strict is not something to "support", it's just something that gets set on the cookie and the browser handles.

> Finally perhaps linking to the change in WebOb would be useful?

I don't think that's useful.

**Comment by bertjwregeer** [comment]
@mcdonc please run the following commands:

```
git checkout master
git fetch --all
git stash -u
git reset --hard origin/master
```

So next time you branch from `master` it doesn't contain all of the merges you've made in the mean time.

**Comment by AvnerCohen** [comment]
Is this expected to be released any time soon ?

**Comment by stevepiercy** [comment]
@AvnerCohen no hard date yet. Progress on the release of [Pyramid 1.10 is tracked in its milestone](https://github.com/Pylons/pyramid/milestone/7). The four "docs" tagged issues are nearly complete or will be dropped from the milestone. Other issues are in progress. Feedback is sought.

**Pull request merged**